builder/amazon/common/ami_config.go

```diff
@@ -20,14 +20,17 @@ type AMIConfig struct {
 	AMIEnhancedNetworking   bool              `mapstructure:"enhanced_networking"`
 	AMIForceDeregister      bool              `mapstructure:"force_deregister"`
 	AMIEncryptBootVolume    bool              `mapstructure:"encrypt_boot"`
+	AMISkipRegister         bool              `mapstructure:"skip_register_ami"`
 }
 
 func (c *AMIConfig) Prepare(ctx *interpolate.Context) []error {
 	var errs []error
-	if c.AMIName == "" {
+	if c.AMIName == "" && !c.AMISkipRegister {
```

Comment:
Sorry for the lag in following up with this.  I think the rest of the validations in the prepare function can currently be skipped.  I put a temporary note below there in the commit that I have to do additional validation that certain other params aren't supplied that don't have any effect with skip_register_ami, though.

Looks like those line references changed since you referenced them, but I presume you mean StepPreValidate and StepSourceAMIInfo.  I suppose I got lucky that the PreValidate does still work, but it should be skipped.  I don't see any reference to the AMI name in StepSourceAMIInfo.  Did you mean a different step?

It looks like StepPreValidate already has ForceDeregister passed in which triggers an informational message.  I'm not sure if that is a convention, but I could wrap the whole step like the others or pass a flag in.  I don't mind either way, but for the larger blocks it seemed to make more sense to do it outside.